diff --git a/BUILD.gn b/BUILD.gn
index 8b768835..3331bf36 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -50,22 +50,11 @@ group("libyuv") {
   all_dependent_configs = [ ":libyuv_config" ]
   deps = []
 
-  if (is_win && target_cpu == "x64") {
-    if (current_os == "winuwp") {
-      public_deps = [
-        ":libyuv_internal",
-      ]
-    } else {
-      # Compile with clang in order to get inline assembly
-      public_deps = [
-        ":libyuv_internal(//build/toolchain/win:win_clang_x64)",
-      ]
-    }
-  } else {
-    public_deps = [
-      ":libyuv_internal",
-    ]
-  }
+  # Always compile with msvc, even if missing inline assembly,
+  # as clang libs won't link with msvc libs.
+  public_deps = [
+    ":libyuv_internal",
+  ]
 
   if (libyuv_use_neon) {
     deps += [ ":libyuv_neon" ]
